### PR TITLE
taskwarrior: 2.5.2 -> 2.5.3

### PR DESCRIPTION
**Pull request by nicknovitski**
###### Motivation for this change

- updated to new version
- run test suite in check phase
- enable for bsd, etc

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after) (2k decrease)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/taskwarrior/default.nix [hunk 1]
```diff
@@ -1,19 +1,33 @@
-{ lib, stdenv, fetchFromGitHub, cmake, libuuid, gnutls }:
-
+{ lib, stdenv, fetchurl, cmake, libuuid, gnutls, python3 }:
+let
+  version = "2.5.3";
+  sourceRoot = "task-${version}";
+in
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/taskwarrior/default.nix:5]:
```suggestion

```

File: pkgs/applications/misc/taskwarrior/default.nix [hunk 2]
```diff
@@ -1,19 +1,33 @@
...
 stdenv.mkDerivation rec {
   pname = "taskwarrior";
-  version = "2.5.2";
+  inherit version sourceRoot;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/taskwarrior/default.nix:8]:
```suggestion
  version = "2.5.3";

  sourceRoot = "task-${version}";
```

**Comment by SuperSandro2000** [comment]
Also the test fails because the program behind ``/usr/bin/env`` in run_all script is not available.

**Comment by SuperSandro2000** [comment]
> Also the test fails because the program behind `/usr/bin/env` in run_all script is not available.

That is still not fixed.

**Comment by nicknovitski** [comment]
> > Also the test fails because the program behind `/usr/bin/env` in run_all script is not available.
> 
> That is still not fixed.

Yeah, there's actually a bunch of files in the test folder with usr/bin/env shebangs and were only working because I'm not using sandboxes yet.  I'll re-request review when I think I have it fixed.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113056` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
13 packages built:
</summary>
<ul>
<li>gnomeExtensions.taskwhisperer</li>
<li>ptask</li>
<li>python37Packages.bugwarrior</li>
<li>python37Packages.tasklib</li>
<li>python37Packages.taskw</li>
<li>python38Packages.bugwarrior</li>
<li>python38Packages.tasklib</li>
<li>python38Packages.taskw</li>
<li>python39Packages.tasklib</li>
<li>python39Packages.taskw</li>
<li>tasknc</li>
<li>taskwarrior</li>
<li>vit</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
gnomeExtensions.taskwhisperer:
<details>


warning: unused-argument
Unused argument: `gnome3`.
Near pkgs/desktops/gnome-3/extensions/taskwhisperer/default.nix:1:84:

```
  |
1 | { lib, stdenv, substituteAll, fetchFromGitHub, taskwarrior, gettext, runtimeShell, gnome3 }:
  |                                                                                    ^
```
</details>
</details>

**Pull request merged**